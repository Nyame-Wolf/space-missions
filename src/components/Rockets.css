.rockets-container {
  margin-top: 110px;
  margin-bottom: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 90%;
  color: #000;
  font-family: var(--roboto);
}

.rockets-list {
  width: 100%;
}

.rockets-list li + li {
  margin-top: 1rem;
}

.rocket-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.rocket-img {
  width: 20%;
  height: 100%;
  flex: 3;
}

.text-style {
  flex: 6;
  margin-left: 1rem;
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
}

.text-style h3 {
  font-weight: 800;
  font-size: 2rem;
  text-align: center;
  padding: 1rem;
}

.text-style p {
  line-height: 1.2;
  font-size: 1.2rem;
  font-weight: 400;
  font-family: var(--monserrat);
  word-spacing: 2px;
  text-align: justify;
  padding: 1rem 1.2rem;
}

.reserve,
.cancel-rocket {
  cursor: pointer;
  width: 12rem;
  height: 2.2rem;
  border-radius: 3px;
  background-color: #0290ff;
  font-family: var(--roboto);
  font-size: 0.85rem;
  font-weight: 400;
  letter-spacing: 0.5px;
  text-align: center;
  color: #fff;
}

.reserve:hover {
  background-color: #43a8f5;
}

.cancel-rocket {
  color: #c60101;
  background-color: #efcccc;
  border: 1px solid #4b0101;
}

.reserved-success {
  background: #28a745;
  color: #000;
  border: none;
  padding: 3px 5px;
  margin: 0;
}


@media screen and (max-width: 650px) {
  .rockets-container {
    width: 100%;
  }

  .rocket-container {
    flex-direction: column;
  }

  .rocket-img {
    width: 100%;
  }

  .text-style {
    margin: 0;
    justify-content: center;
  }
}
