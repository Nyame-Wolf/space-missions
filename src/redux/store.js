import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';

import rocketReducer from './rockets/rockets';

import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
