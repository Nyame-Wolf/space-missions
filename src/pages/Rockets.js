import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Rocket from '../components/Rocket';
import { getRockets } from '../redux/rockets/rockets';
import '../components/Rockets.css';

export default function Rockets() {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets);

  useEffect(() => {
    if (!rockets.length) {
      dispatch(getRockets());
    }
  }, []);

  return (

    <div className="rockets-container">
      <ul className="rockets-list">
        {rockets.map((rocket) => (
          <li key={rocket.id}>
            <Rocket rocket={rocket} />
          </li>
        ))}
      </ul>
    </div>
  );
}
