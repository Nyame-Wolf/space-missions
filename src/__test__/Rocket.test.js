import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../pages/Rockets';
import store from '../redux/store';

describe('Rockets test', () => {
  test('snapshot for rockets', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
