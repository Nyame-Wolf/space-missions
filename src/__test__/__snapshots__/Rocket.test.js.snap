// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets test snapshot for rockets 1`] = `
<div
  className="rockets-container"
>
  <ul
    className="rockets-list"
  />
</div>
`;
